README.md
# Description

The site from Proyectos Beta

## Tecnologies

- HTML 5
- NodeJS v18.12.1
- NPM 8.19.2
- React 18.2.0
- Gatsby 5

## Developer

```bash
    npm i
    gatsby develop
```

## Production

```bash
    gatsby build
```


CHANGELOG.md
# Release Notes

## v1.2.0

- [feat] New version site. Using gatsby 5.
- [docs] Update changelog file.

## v1.1.0

- [fix] Font Google Open Sans, size larger and change color.
- [feat] Add link for the Wiki in the header.

## v1.0.0

- [feat] Add site Proyectos Beta.
- [docs] Update readme file.
- [feat] Add telegram social media.
- [fix] Hide the header.

gatsby-config.js
/**
 * Configure your Gatsby site with this file.
 *
 * See: https://www.gatsbyjs.com/docs/reference/config-files/gatsby-config/
 */

require("dotenv").config();

/**
 * @type {import('gatsby').GatsbyConfig}
 */
module.exports = {
  siteMetadata: {
    title: `Proyectos Beta`,
    description: `Y el todo en nuestra vida es un Proyecto. Siempre podemos mejorar porque estamos en fase Beta..`,
    author: `Proyectos Beta TEAM`,
    siteUrl: `https://proyectosbeta.net/`,
    social: {
      twitter: `proyectosbeta`,
    },
    socialLinks: [
      {
        name: "facebook",
        url: "https://www.facebook.com/proyectosbeta1",
      },
      {
        name: "twitter",
        url: "https://twitter.com/proyectosbeta",
      },
      {
        name: "youtube",
        url: "https://www.youtube.com/proyectosbeta",
      },
      {
        name: "telegram",
        url: "https://telegram.me/proyectosbeta",
      },
      {
        name: "instagram",
        url: "https://www.instagram.com/proyectosbeta1/",
      },
      {
        name: "github",
        url: "https://github.com/proyectosbeta",
      },
    ],
  },
  plugins: [
    `gatsby-plugin-image`,
    {
      resolve: `gatsby-source-filesystem`,
      options: {
        name: `images`,
        path: `${__dirname}/src/images`,
      },
    },
    `gatsby-transformer-sharp`,
    `gatsby-plugin-sharp`,
    {
      resolve: `gatsby-plugin-manifest`,
      options: {
        name: `gatsby-starter-default`,
        short_name: `starter`,
        start_url: `/`,
        background_color: `#663399`,
        // This will impact how browsers show your PWA/website
        // https://css-tricks.com/meta-theme-color-and-trickery/
        // theme_color: `#663399`,
        display: `minimal-ui`,
        icon: `src/images/favicon.png`, // This path is relative to the root of the site.
      },
    },
    {
      resolve: `gatsby-plugin-google-gtag`,
      options: {
        trackingIds: [
          process.env.GOOGLE_ANALYTICS_ID, // Google Analytics
        ],
        gtagConfig: {
          optimize_id: "OPT_CONTAINER_ID",
          anonymize_ip: true,
          cookie_expires: 0,
        },
        pluginConfig: {
          head: true,
          respectDNT: true,
        },
      },
    },
  ],
};


src/pages/index.js
import * as React from "react";
import { StaticImage } from "gatsby-plugin-image";
import Layout from "../components/layout";
import Seo from "../components/seo";

const IndexPage = () => (
  <Layout>
    <center>
      <h1>
        Bienvenidos a <b>Proyectos Beta!</b>
      </h1>
      <div>
        Estamos de Mantenimiento!!!
        <StaticImage
          src="../images/construction.png"
          alt="Sitio en mantenimieto"
          placeholder="Sitio en mantenimieto"
          layout="fixed"
          width={300}
          height={300}
        />
      </div>
    </center>
  </Layout>
);

/**
 * Head export to define metadata for the page
 *
 * See: https://www.gatsbyjs.com/docs/reference/built-in-components/gatsby-head/
 */
export const Head = () => <Seo title="Home" />;

export default IndexPage;


src/components/footer.js
import React from "react";
import styled from "styled-components";
import Container from "./container";
import SocialLinks from "./social-links";

const StyledFooter = styled.footer`
  padding-top: var(--size-300);
  padding-bottom: var(--size-300);
`;

const FooterAttribution = styled.p`
  font-size: var(--size-300);

  & a {
    color: inherit;
  }
`;

const FooterWrapper = styled(Container)`
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  align-items: center;
  background: #e9e9e9;
  color: #252526;
`;

const Footer = () => {
  return (
    <StyledFooter>
      <FooterWrapper>
        <SocialLinks />

        <FooterAttribution>
          © 2023 Copyright: Proyectos Beta
        </FooterAttribution>
      </FooterWrapper>
    </StyledFooter>
  );
};

export default Footer;


src/components/layout.js
import React, { Fragment } from "react";
import styled from "styled-components";
// import Seo from "./seo";
import Header from "./header";
import Footer from "./footer";
import Container from "./container";
import GlobalStyle from "./global-styles";

const LayoutWrapper = styled.div`
  min-height: 100vh;
  display: flex;
  flex-direction: column;

  & main {
    margin-top: auto;
    margin-bottom: auto;
  }

  & footer {
    margin-top: auto;
  }
`;

const Layout = ({ children, title, description, socialImage = "" }) => {
  return (
    <Fragment>
      <GlobalStyle />
      {/* <Seo title={title} description={description} socialImage={socialImage} /> */}
      <LayoutWrapper>
        <Header />
        <main>
          <Container>{children}</Container>
        </main>
        <Footer />
      </LayoutWrapper>
    </Fragment>
  );
};

export default Layout;


src/components/social-links.js
import React from "react";
import { graphql, useStaticQuery } from "gatsby";
import styled from "styled-components";

const SocialLinkList = styled.ul`
  padding: 0;
  list-style: none;
  display: flex;
`;

const SocialLinkItem = styled.li`
  margin-right: var(--size-400);
  text-transform: uppercase;

  & a {
    color: inherit;
    font-size: var(--size-300);
  }
`;

const SocialLinks = () => {
  const data = useStaticQuery(graphql`
    {
      site {
        siteMetadata {
          socialLinks {
            name
            url
          }
        }
      }
    }
  `);

  const socialLinks = data.site.siteMetadata.socialLinks.map(link => {
    return (
      <SocialLinkItem key={link.name}>
        <a href={link.url}>{link.name}</a>
      </SocialLinkItem>
    );
  });

  return <SocialLinkList>{socialLinks}</SocialLinkList>;
};

export default SocialLinks;


src/components/global-styles.js
import { createGlobalStyle } from "styled-components";

const GlobalStyle = createGlobalStyle`
*,
*::before,
*::after {
  box-sizing: border-box;
}

body,
h1,
h2,
h3,
h4,
p,
figure,
blockquote,
dl,
dd {
  margin: 0;
}

ul[role='list'],
ol[role='list'] {
  list-style: none;
}

html:focus-within {
  scroll-behavior: smooth;
}

html {
  height: -webkit-fill-available;
}

body {
  min-height: 100vh;
  min-height: -webkit-fill-available;
  text-rendering: optimizeSpeed;
  line-height: 1.5;
}

a:not([class]) {
  text-decoration-skip-ink: auto;
}

img,
picture {
  max-width: 100%;
  display: block;
}

input,
button,
textarea,
select {
  font: inherit;
}

@media (prefers-reduced-motion: reduce) {
  html:focus-within {
   scroll-behavior: auto;
  }
  
  *,
  *::before,
  *::after {
    animation-duration: 0.01ms !important;
    animation-iteration-count: 1 !important;
    transition-duration: 0.01ms !important;
    scroll-behavior: auto !important;
  }
}

html {
  width:100vw;
  overflow-x:hidden;
}

body::-webkit-scrollbar {
    width: 10px;
}

body::-webkit-scrollbar-thumb {
  background-color: rgba(55, 41, 44, .4);
}

body::-webkit-scrollbar-track {
    background: transparent;
}

:root {
  scroll-behavior: smooth;
  --size-300: 0.75rem;
  --size-400: 1rem;
  --size-500: 1.33rem;
  --size-600: 1.77rem;
  --size-700: 2.36rem;
  --size-800: 3.15rem;
  --size-900: 4.2rem;
}

body {
  font-family: 'Poppins', -apple-system, BlinkMacSystemFont, avenir next, avenir, helvetica neue,
    helvetica, Ubuntu, roboto, noto, segoe ui, arial, sans-serif;
  transition-property: background-color, color, background;
  transition-duration: 0.3s;
  transition-timing-function: ease-out;
  background-attachment: fixed;
  font-smoothing: antialiased;
}

body {
  color: #e9e9e9;
  background-color: #252526;
}

body ::selection{
  background: #e9e9e9;
  color: #252526;
}

h1,
h2,
h3,
h4 {
  line-height: 1.125;
}

h1,
h2,
h3 {
  font-weight: 700;
}

h1 {
  font-size: var(--size-800);
}

h2 {
  font-size: var(--size-700);
}

h3 {
  font-size: var(--size-600);
}

p {
  font-size: var(--size-400);
}

p, li {
    max-width: none;
}

.gatsby-resp-image-wrapper {
    margin-left: 0 !important;
}


`;

export default GlobalStyle;


src/components/header.js
import * as React from "react";
import styled from "styled-components";
import { Link } from "gatsby";
import Container from "./container";
import { useStaticQuery, graphql } from "gatsby";

const HEADER_NAV_ITEM = [
  {
    label: "Wiki",
    url: "https://wiki.proyectosbeta.net",
    isExternal: true,
  },
  // {
  //   label: "About",
  //   url: "/about",
  //   isExternal: false,
  // },
  // {
  //   label: "Contact",
  //   url: "/contact",
  //   isExternal: false,
  // },
];

const HeaderNavList = ({ children }) => {
  return (
    <StyledNav>
      <StyledNavList>{children}</StyledNavList>
    </StyledNav>
  );
};

const HeaderNavListItem = ({ children }) => {
  return <StyledNavListItem>{children}</StyledNavListItem>;
};

const StyledHeader = styled.header`
  padding-top: var(--size-300);
`;

const HeaderWrapper = styled(Container)`
  display: flex;
  align-items: center;
  justify-content: space-between;
`;

const HeaderTitle = styled.div`
  & a {
    text-transform: uppercase;
    text-decoration: none;
    font-size: var(--size-400);
    color: inherit;
  }
  color: red;
`;

const StyledNav = styled.nav`
  position: static;
  padding: 0;
  background: transparent;
  backdrop-filter: unset;
`;

const StyledNavList = styled.ul`
  display: flex;
  align-items: center;
  flex-wrap: wrap;
  justify-content: space-around;
  padding: 0;
  list-style-type: none;
`;

const StyledNavListItem = styled.li`
  &:not(:last-of-type) {
    margin-right: 2rem;
  }
  @media screen and (max-width: 700px) {
    &:not(:last-of-type) {
      margin-right: 1rem;
    }
  }
  & a {
    color: inherit;
    text-transform: uppercase;
    font-size: var(--size-300);
    text-decoration: none;
    letter-spacing: 0.1rem;
  }
  @media screen and (max-width: 700px) {
    & a {
      font-size: 0.7rem;
    }
  }
`;

const Header = () => {
  const { site } = useStaticQuery(
    graphql`
      query {
        site {
          siteMetadata {
            title
          }
        }
      }
    `
  );

  return (
    <StyledHeader>
      <HeaderWrapper>
        <HeaderTitle>
          <Link to="/">{site.siteMetadata.title}</Link>
        </HeaderTitle>

        <HeaderNavList>
          {HEADER_NAV_ITEM.map((item, index) => {
            if (item.isExternal) {
              return (
                <HeaderNavListItem key={index}>
                  <a href={item.url} target="_blank" rel="noopener noreferrer">
                    {item.label}
                  </a>
                </HeaderNavListItem>
              );
            }

            return (
              <HeaderNavListItem key={index}>
                <Link to={item.url}>{item.label}</Link>
              </HeaderNavListItem>
            );
          })}
        </HeaderNavList>
      </HeaderWrapper>
    </StyledHeader>
  );
};

export default Header;
